src/test/regress/expected/multi_index_statements.out

```diff
@@ -292,13 +296,92 @@ SELECT indisvalid AS "Index Valid?" FROM pg_index WHERE indexrelid='ith_b_idx'::
+
+-- should be able to create short named indexes in parallel
+-- as there are no partitions even if the index name is too long
+SET client_min_messages TO DEBUG1;
+CREATE INDEX ix_test_index_creation1_ix_test_index_creation1_ix_test_index_creation1_ix_test_index_creation1_ix_test_index_creation1
+    ON test_index_creation1 USING btree
+    (tenant_id, timeperiod);
+NOTICE:  identifier "ix_test_index_creation1_ix_test_index_creation1_ix_test_index_creation1_ix_test_index_creation1_ix_test_index_creation1" will be truncated to "ix_test_index_creation1_ix_test_index_creation1_ix_test_index_c"
+RESET client_min_messages;
+CREATE TABLE test_index_creation1_p2020_09_26 PARTITION OF test_index_creation1 FOR VALUES FROM ('2020-09-26 00:00:00') TO ('2020-09-27 00:00:00');
+CREATE TABLE test_index_creation1_p2020_09_27 PARTITION OF test_index_creation1 FOR VALUES FROM ('2020-09-27 00:00:00') TO ('2020-09-28 00:00:00');
+-- should switch to sequential execution as the index name on the partition is
+-- longer than 63
+SET client_min_messages TO DEBUG1;
+CREATE INDEX ix_test_index_creation2
+    ON test_index_creation1 USING btree
+    (tenant_id, timeperiod);
+DEBUG:  the index name on the shards of the partition is too long, switching to sequential execution mode to prevent self deadlocks: test_index_creation1_p2020_09_26_10311_tenant_id_timeperiod_idx
+-- same test with schema qualified
+SET search_path TO public;
+CREATE INDEX ix_test_index_creation3
+    ON multi_index_statements.test_index_creation1 USING btree
+    (tenant_id, timeperiod);
+DEBUG:  the index name on the shards of the partition is too long, switching to sequential execution mode to prevent self deadlocks: test_index_creation1_p2020_09_26_10311_tenant_id_timeperiod_idx
+SET search_path TO multi_index_statements;
+-- we cannot switch to sequential execution
+-- after a parallel query
+BEGIN;
+	SELECT count(*) FROM test_index_creation1;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+	CREATE INDEX ix_test_index_creation4
+	ON test_index_creation1 USING btree
+	(tenant_id, timeperiod);
+ERROR:  The index name (test_index_creation1_p2020_09_26_10311_tenant_id_timeperiod_idx) on a shard is too long and could lead to deadlocks when executed in a transaction block after a parallel query
+HINT:  Try re-running the transaction with "SET LOCAL citus.multi_shard_modify_mode TO 'sequential';"
+ROLLBACK;
+-- try inside a sequential block
+BEGIN;
+	SET LOCAL citus.multi_shard_modify_mode TO 'sequential';
+	SELECT count(*) FROM test_index_creation1;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+	CREATE INDEX ix_test_index_creation4
+	ON test_index_creation1 USING btree
+	(tenant_id, timeperiod);
+DEBUG:  the index name on the shards of the partition is too long, switching to sequential execution mode to prevent self deadlocks: test_index_creation1_p2020_09_26_10311_tenant_id_timeperiod_idx
```

Comment:
would it be worth using a NOTICE? not sure whether it will inform or scare the user

Reply:
We do not have `NOTICE` when we switch to sequential execution due to fkey relationships. That's why I followed the same approach by adding DEBUG1.

If table is empty, NOTICE  would be confusing. If the table has reasonable amount of data, then it might be beneficial.

Do we have a trick to get that information without calling `master_update_shard_stats`?